ouroboros-network-framework/test/Test/Ouroboros/Network/Server2.hs

```diff
@@ -2116,159 +2041,145 @@ prop_multinode_Sim serverAcc (ArbDataFlow dataFlow) absBi script@(MultiNodeScrip
 --
 prop_multinode_Sim_Pruning_Transitions :: Int -> Small Word32 -> MultiNodeScriptPruning Int -> Property
 prop_multinode_Sim_Pruning_Transitions serverAcc (Small bound) (MultiNodeScriptPruning l) =
-  let trace = runSimTrace sim
-        where
-          sim :: IOSim s ()
-          sim = do
-            mb <- timeout 7200
-                    ( withSnocket sayTracer
-                                  (singletonScript noAttenuation)
-                    $ \snocket ->
-                       multinodeExperiment (Tracer traceM <> debugTracer)
-                                           (Tracer traceM <> debugTracer)
-                                           (Tracer traceM <> debugTracer)
-                                           (Tracer traceM <> debugTracer)
-                                           snocket
-                                           Snocket.TestFamily
-                                           (Snocket.TestAddress 0)
-                                           serverAcc
-                                           Duplex
-                                           (bound `div` 10)
-                                           (unTestAddr <$> MultiNodeScript l)
-                    )
-            case mb of
-              Nothing -> throwIO (SimulationTimeout :: ExperimentError SimAddr)
-              Just a  -> return a
-      evs :: Octopus (Value ()) (AbstractTransitionTrace SimAddr)
+  let evs :: Octopus (Value ())
```

Comment:
> The other remark about scope of validation also applies here. It will be much simpler to only validate the pruning, without re-validating all the other properties.

I am not sure I understand

Reply:
We test three things here, where I think we should have three tests that check each property separetly:

1.
>               -- verify that 'unregisterInboundConnection' does not return
>               -- 'UnsupportedState'.
>               ...
>               -- verify that 'demotedToColdRemote' does not return
>               -- 'UnsupportedState'

2. `mkPropertyPruning`
3. `verifyRemoteTransition`

Reply:
I'll leave this change for this commit: https://github.com/input-output-hk/ouroboros-network/pull/3360/commits/4b88e40184d5e55c640489f4cfdf8cbb77953f2f